### PR TITLE
kernel: allowlist v2

### PR DESCRIPTION
**Pull request by Ylarod**

### CODE REVIEW

File: kernel/allowlist.c [hunk 1]
```diff
@@ -1,25 +1,40 @@
+#include "asm-generic/int-ll64.h"
```
Comment (tiann) [inline kernel/allowlist.c:1]: It this we need?

File: kernel/allowlist.c [hunk 2]
```diff
@@ -1,25 +1,40 @@
...
 #include "selinux/selinux.h"
 
 #define FILE_MAGIC 0x7f4b5355 // ' KSU', u32
-#define FILE_FORMAT_VERSION 1 // u32
+#define FILE_FORMAT_VERSION 2 // u32
 
 static DEFINE_MUTEX(allowlist_mutex);
 
-struct perm_data {
+const struct perm_data NO_PERM = {
```
Comment (tiann) [inline kernel/allowlist.c:20]: maybe static?

File: kernel/allowlist.c [hunk 3]
```diff
@@ -1,25 +1,40 @@
...
+	.allow = false,
+	.cap = CAP_EMPTY_SET,
+};
+
+const struct perm_data ALL_PERM = {
```
Comment (tiann) [inline kernel/allowlist.c:25]: static

File: kernel/allowlist.c [hunk 4]
```diff
@@ -28,42 +43,56 @@ static struct work_struct ksu_load_work;
 
 bool persistent_allow_list(void);
 
+void ksu_show_perm_data(struct perm_data data)
```
Comment (tiann) [inline kernel/allowlist.c:46]: use `struct perm_data *`

File: kernel/allowlist.c [hunk 5]
```diff
@@ -28,42 +43,56 @@ static struct work_struct ksu_load_work;
...
+{
+	pr_cont("allow: %d, cap: %08x | %08x", data.allow, data.cap.cap[0],
+		data.cap.cap[1]);
+}
+
+void ksu_show_perm_list_node(const char* str, struct perm_list_node data)
```
Comment (tiann) [inline kernel/allowlist.c:52]: use pointer

File: kernel/allowlist.c (hunk 4 above)
Comment (tiann) [reply kernel/allowlist.c:46]: and static

File: kernel/allowlist.c (hunk 5 above)
Comment (tiann) [reply kernel/allowlist.c:52]: static

Comment (tiann) [reply kernel/allowlist.c:52]: rename it to node to avoid `data->data`

File: kernel/allowlist.c [hunk 6]
```diff
@@ -73,49 +102,74 @@ bool ksu_allow_uid(uid_t uid, bool allow, bool persist)
 	return result;
 }
 
-bool ksu_is_allow_uid(uid_t uid)
+struct perm_data ksu_get_uid_data(uid_t uid)
```
Comment (tiann) [inline kernel/allowlist.c:105]: `ksu_get_uid_perm` is better, so many "data" here, it is confusing

**Review by tiann — CHANGES_REQUESTED** [review]

File: kernel/allowlist.c (hunk 2 above)
Comment (Ylarod) [reply kernel/allowlist.c:20]: exposed to other part, cannot be static

**Pull request closed without merging**